src/v/cluster/archival/tests/segment_reupload_test.cc

```diff
@@ -843,19 +844,18 @@ SEASTAR_THREAD_TEST_CASE(test_upload_aligned_to_non_existent_offset) {
               .get();
         }
         auto seg = b.get_log_segments().back();
-        seg->appender().close().get();
-        seg->release_appender().get();
+        seg->release_appender(&b.get_disk_log_impl().readers()).get();
```

Comment:
This `release_appender(readers_cache)` call releases both the segment appender and the compacted index appender (this method is what's used in disk_log_impl). This is now a requirement for this test because the merge compaction in `gc()` will close a compacted index if it hasn't yet been released, and that would trip up subsequent `segment->close()` calls, which expect to only be called once.